sources/packages/api/src/esdc-integration/e-cert-integration/e-cert-part-time-integration/e-cert-files/e-cert-file-record.ts

```diff
@@ -0,0 +1,193 @@
+import { StringBuilder } from "../../../../utilities/string-builder";
+import { RecordTypeCodes } from "../models/e-cert-part-time-integration.model";
+import {
+  SPACE_FILLER,
+  NUMBER_FILLER,
+  DATE_FORMAT,
+} from "../../../models/esdc-integration.model";
+
+/**
+ * Record of an Entitlement E-Cert file.
+ * The documentation about it is available on the document
+ * 'CSLP-AppendixF2AsReviewed2016-FileLayouts BC Files V3(HAJ-CB EDITS) In ESDC Folder'.
+ */
+export class ECertPTFileRecord implements FixedFormatFileLine {
+  recordType: RecordTypeCodes;
+  /**
+   * Social insurance number of student.
+   */
+  sin: string;
+  /**
+   * Date when the funds on this document (e-Cert) can be released.
+   */
+  disbursementDate: Date;
+  /**
+   * Date the document (e-Cert) was created.
+   */
+  documentProducedDate: Date;
+  /**
+   * Total dollar amount of all funding types on this document (e-Cert).
+   * ! Must be rounded, only integer, no decimals.
+   */
+  disbursementAmount: number;
+  /**
+   * Date that studies commence. Generally derived from last application. The first day of classes.
+   * Also known as the Period of Study Commencement Date (PSCD).
+   */
+  educationalStartDate: Date;
+  /**
+   * Date that studies end. Generally derived from the latest application. Last day of classes or last
+   * exam date, whichever is latest. Also known as the Period of Study End Date (PSED).
+   */
+  educationalEndDate: Date;
+  /**
+   * Federal school code.
+   */
+  federalInstitutionCode: string;
+  /**
+   * Number of study weeks in study period.
+   */
+  weeksOfStudy: number;
+  /**
+   * Federal field of study code.
+   */
+  fieldOfStudy: number;
+  /**
+   * Student's Year of study.
+   */
+  yearOfStudy: number;
+  /**
+   * Date that the school confirmed enrolment.
+   */
+  enrollmentConfirmationDate: Date;
+  /**
+   * Student date of birth.
+   */
+  dateOfBirth: Date;
+  /**
+   * Student last name.
+   */
+  lastName: string;
+  /**
+   * Student first name.
+   */
+  firstName: string;
+  /**
+   * Student address.
+   */
+  addressLine1: string;
+  /**
+   * Student address
+   */
+  addressLine2: string;
+  /**
+   * Student city address.
+   */
+  city: string;
+  /**
+   * Student email address.
+   */
+  emailAddress: string;
+  /**
+   * Student gender M=Male F=Female.
+   */
+  gender: string;
+  /**
+   * Borrower marital status S= Single, M = Married, O = Other.
+   */
+  maritalStatus: string;
+  /**
+   * Student number (at the Educational Institution).
+   */
+  studentNumber: string;
+  /**
+   * Total of all grant awards on this Entitlement Record.
+   */
+  totalGrantAmount: number;
+  /**
+   * Certificate number, the generated sequence number for the file.
+   */
+  certNumber: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalBCSGAmount: number;
+  /**
+   * Total of all BC grant awards on this Entitlement Record.
+   */
+  totalCSGP_PTAmount: number;
```

Comment:
Please rename it to _totalCSGPPartTimeAmount_.

Reply:
This is how we have it in our workflow, so followed it.